src/throttler/throttler.ts

```diff
@@ -41,14 +47,13 @@ export abstract class Throttler<T> {
   public errored: number = 0;
   public retried: number = 0;
   public total: number = 0;
-  public tasks: { [index: number]: T } = {};
+  public taskDatas: { [index: number]: TaskData<T> } = {};
```

Comment:
I'm not thrilled about `taskDatas` - maybe `taskDataMap`?

Also, while we're at it: we're in ES6! Use [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)! We shouldn't be using `{}` as much as we are...